### PR TITLE
CA-257927: Fixed scalability issue where XenCenter froze momentarily when disconnecting from a host with a big number of alerts

### PR DESCRIPTION
**Pull request by kc284**

### CODE REVIEW

File: XenAdmin/TabPages/AlertSummaryPage.cs [hunk 1]
```diff
@@ -585,24 +584,27 @@ private void tsmiDismissSelected_Click(object sender, EventArgs e)
...
-                    RemoveAlertRow(a);
+
+                    var a = e.Element as Alert;
+                    if (a != null)
+                        RemoveAlertRow(a);
+                    else
+                    {
+                        var range = e.Element as List<Alert>;
+                        if (range != null)
+                            Rebuild();
```
Comment (MihaelaStoica) [inline XenAdmin/TabPages/AlertSummaryPage.cs]: I wonder if it's worth checking if there was just one element in the list, and in this case call RemoveAlertRow?

**Comment by MihaelaStoica** [comment]
@kc284 Apart for the question above, these changes look good to me.

File: XenAdmin/TabPages/AlertSummaryPage.cs (hunk 1 above)
Comment (kc284) [inline XenAdmin/TabPages/AlertSummaryPage.cs]: Can do. Hmm, or alternatively put the check in the Changeable list and fire the event for a single item in line 87?

Comment (MihaelaStoica) [inline XenAdmin/TabPages/AlertSummaryPage.cs]: I like the idea of putting the check in the Changeable list

**Comment by MihaelaStoica** [comment]
Looks good. Merging.

**Pull request merged**